driver/src/main/java/com/code_intelligence/jazzer/junit/RegressionTestArgumentProvider.java

```diff
@@ -0,0 +1,135 @@
+// Copyright 2022 Code Intelligence GmbH
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.code_intelligence.jazzer.junit;
+
+import static org.junit.jupiter.api.Named.named;
+import static org.junit.jupiter.params.provider.Arguments.arguments;
+
+import com.code_intelligence.jazzer.api.FuzzedDataProvider;
+import com.code_intelligence.jazzer.driver.FuzzedDataProviderImpl;
+import java.io.File;
+import java.io.IOException;
+import java.lang.reflect.Method;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.net.URL;
+import java.nio.file.FileSystem;
+import java.nio.file.FileSystems;
+import java.nio.file.FileVisitOption;
+import java.nio.file.Files;
+import java.nio.file.NoSuchFileException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.AbstractMap.SimpleEntry;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.jar.JarEntry;
+import java.util.jar.JarFile;
+import java.util.jar.JarInputStream;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import org.junit.jupiter.api.extension.ExtensionContext;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.ArgumentsProvider;
+import org.junit.jupiter.params.support.AnnotationConsumer;
+
+class RegressionTestArgumentProvider implements ArgumentsProvider, AnnotationConsumer<FuzzTest> {
+  FuzzTest annotation;
+
+  @Override
+  public void accept(FuzzTest annotation) {
+    this.annotation = annotation;
+  }
+
+  @Override
+  public Stream<? extends Arguments> provideArguments(ExtensionContext extensionContext)
+      throws IOException {
+    Stream<Map.Entry<String, byte[]>> rawSeeds =
+        Stream.concat(Stream.of(new SimpleEntry<>("<empty input>", new byte[] {})),
+            walkSeedCorpus(extensionContext.getRequiredTestClass()));
+    return adaptSeedsForFuzzTest(extensionContext.getRequiredTestMethod(), rawSeeds);
+  }
+
+  private Stream<? extends Arguments> adaptSeedsForFuzzTest(
+      Method fuzzTestMethod, Stream<Map.Entry<String, byte[]>> rawSeeds) {
+    if (fuzzTestMethod.getParameterCount() != 1) {
+      throw new IllegalArgumentException();
```

Comment:
Use a descriptive message like in the case of a wrong parameter type? Also a test for this case would be great :-)

Reply:
Added the same message and a test.